### PR TITLE
v0.3.0 release

### PR DESCRIPTION
**Pull request by vpnagraj**
this release brings in the following changes:

- `who_nrevss()` helper and `get_cdc_clin()` function for retrieving clinical laboratory positive flu data
- `clin_nowcast()` engine
- `smoothie()` helper for calculating weighted averages
- `pois_forc()` to forecast counts for horizons ahead based on characteristics of observed series
- update to submission.R to allow SigSci-CREG to use laboratory positivity and "autoregressive" covariates

### CODE REVIEW

File: R/forecast.R [hunk 1]
```diff
@@ -359,3 +359,78 @@ forecast_ili <- function(ilidat, horizon=4L, trim_date=NULL, models=list(arima="
...
+#' Simple Poisson count forecaster
+#'
+#' @description This function is a helper that forecasts Poisson counts for 4 near-term horizons based on characteristics of recently observed count data. The function effectively takes a rolling average of last 4 observations (augmenting with each forecasted horizon as the horizons progress), then uses this average as the parameter for Lambda in a random draw from a Poisson distribution.
+#'
+#' @param .data Data frame with incoming data that includes a variable with counts (see ".var" argument), and location (must be stored in a column called "location") and a variable for sorting by date (must be stored in a column called "week_start")
+#' @param .location The name of the location of interest
+#' @param .var Bare, unquoted name of the variable with counts to be forecasted
+#'
+#' @return Vector of length 4 with Poisson forecasts for 4 horizons ahead.
+#' @export
```
Comment (stephenturner) [inline R/forecast.R:412]: maybe add one or two `@examples`?

File: R/forecast.R [hunk 2]
```diff
@@ -359,3 +359,78 @@ forecast_ili <- function(ilidat, horizon=4L, trim_date=NULL, models=list(arima="
...
+#'
+#'
+pois_forc <- function(.data, .location, .var) {
+
+  ## handle for NSE
+  tmp_var <- dplyr::enquo(.var)
```
Comment (stephenturner) [inline R/forecast.R:418]: no need to change, but an FYI, the curlycurly {{ is useful for stuff like this https://www.tidyverse.org/blog/2019/06/rlang-0-4-0/

File: R/retrieve.R [hunk 3]
```diff
@@ -308,6 +308,71 @@ get_nowcast_ili <- function(epiyearweeks=NULL, dates=lubridate::today()-c(14,7),
...
 
+#' Retrieve and prep clinical laboratory percent positive flu data
+#'
+#' @description This function returns weekly state and/or national clinical laboratory percent positivity data from the NREVSS reporting instrument.
+#'
+#' @param region Either "state", "national", or "both". Defaults to "both" to return state and national data combined.
+#' @param years A vector of years to retrieve data for. CDC has data going back to 1997. Default value (`NULL`) retrieves **all** years.
+#'
+#' @return Tibble with prepared data pulled from NREVSS, including columns for total number of positives, percent positive, epiweek, epiyear
+#' @export
```
Comment (stephenturner) [inline R/retrieve.R:319]: add a `@examples` under `\dontrun{}`?

File: R/utils.R [hunk 4]
```diff
@@ -515,3 +515,35 @@ surveillance_areas <- function() {
...
+#' Calculate smoothed and weighted averages of previous observations
+#'
+#' @description This helper function calculates a weighted average of the last n observations.
+#'
+#' @param x Incoming vector of observations
+#' @param n Number of recent observations to smooth; default is `4`
+#' @param weights Vector of weights to be applied to last n observations during averaging
+#'
+#' @return Vector of length 1 with the weighted average of last n observations.
+#' @export
```
Comment (stephenturner) [inline R/utils.R:528]: definitely add a couple `@examples` to this one

**Review by stephenturner — COMMENTED** [review]
looks great, haven't run anything, suggesting adding a few `@examples`

**Comment by vpnagraj** [comment]
good call(s) on examples. added throughout

**Pull request merged**